includes/admin/class-amp-post-meta-box.php

```diff
@@ -422,4 +423,22 @@ public function preview_post_link( $link ) {
 
 		return $link;
 	}
+
+	/**
+	 * Hide the `amp_status` custom field when edition a post.
+	 *
+	 * @param bool   $protected Whether the key is considered protected.
+	 * @param string $meta_key Metadata key.
+	 * @param string $meta_type Type of object metadata is for.
+	 * @return bool
+	 */
+	public function hide_amp_status_meta( $protected, $meta_key, $meta_type ) {
+		global $pagenow;
+
+		if ( is_admin() && 'post.php' === $pagenow && self::STATUS_POST_META_KEY === $meta_key && 'post' === $meta_type ) {
+			$protected = true;
```

Comment:
This is just the hacky way I was able to quickly make this work. However, it seems there should be a more elegant way to expose a setting in the REST API but hide it from the Custom Fields?

Reply:
In my testing the field was still visible in the REST API, unless I'm missing something?

Reply:
Yes, it is still visible. But I mean for postmeta in general. How is this handled for other plugins that expose postmeta that manipulate them via REST API but want to avoid them to be shown in the Custom Fields metabox? There's gotta be a more elegant way to do this.

Reply:
The only way to not render this field would be to mark it as protected ([ref](https://github.com/WordPress/wordpress-develop/blob/0b4e2c46044759e665f7eb516cd4bc5eb37d04d4/src/wp-admin/includes/meta-boxes.php#L755)).

Reply:
@westonruter You are probably thinking about `register_meta()`, but while that can decide whether a meta value should be shown in a REST API response or not (`'show_in_rest'`), it does not have any effect on the post meta box.